File: lib/src/model/bridge/movie_data.dart
import 'package:android_tv_channels/src/model/bridge/aspect_ratio.dart';

class MovieData {
  final String title;
  final int? durationMillis;
  final String intentUri;
  final String? description;
  final DateTime? releaseDate;
  final String? genre;
  final String? posterUri;
  final PosterAspectRatio? posterAspectRatio;

  const MovieData({
    required this.title,
    required this.intentUri,
    this.durationMillis,
    this.description,
    this.releaseDate,
    this.genre,
    this.posterUri,
    this.posterAspectRatio,
  });
}

File: android/src/main/kotlin/be/vanlooverenkoen/androidtvchannels/flutter_android_tv_channels/model/MovieData.kt
package be.vanlooverenkoen.androidtvchannels.flutter_android_tv_channels.model

import java.text.DateFormat
import java.util.Date

data class MovieData(
    val title: String,
    val intentUri: String,
    val durationMillis: Int?,
    val description: String?,
    val releaseDate: Date?,
    val genre: String?,
    val posterUri: String?,
    val posterAspectRatio: String?
)

object MovieDataParser {
    fun parseMovies(data: List<Map<String, Any>>): List<MovieData> = data.map {
        val date = it["releaseDate"] as String?
        MovieData(
            title = it["title"] as String,
            durationMillis = it["durationMillis"] as Int?,
            intentUri = it["intentUri"] as String,
            releaseDate = if (date == null) null else DateFormat.getInstance().parse(date),
            genre = it["genre"] as String?,
            description = it["description"] as String?,
            posterUri = it["posterUri"] as String?,
            posterAspectRatio = it["posterAspectRatio"] as String?
        )
    }
}